### PR TITLE
Support EditInfo parameters in all write services to allow custom summaries and bot edits

### PR DESCRIPTION
**Pull request by Benestar**
Fixes https://phabricator.wikimedia.org/T111773

### CODE REVIEW

File: src/Api/WikibaseFactory.php [hunk 1]
```diff
@@ -116,7 +116,7 @@ public function newValueFormatter() {
 	 * @return ItemMerger
 	 */
 	public function newItemMerger() {
-		return new ItemMerger( $this->api );
+		return new ItemMerger( new WikibaseApi( $this->api ) );
```
Comment (JeroenDeDauw) [inline src/Api/WikibaseFactory.php]: Duplication... if you now change the `WikibaseApi` constructor params, you'll need change stuff here in a dozen places

File: src/Api/Service/DescriptionSetter.php [hunk 2]
```diff
@@ -46,8 +49,7 @@ public function set( Term $description, $target ) {
 		$params['language'] = $description->getLanguageCode();
 		$params['value'] = $description->getText();
 
-		$params['token'] = $this->api->getToken();
```
Comment (JeroenDeDauw) [inline src/Api/Service/DescriptionSetter.php]: Nice you got rid of this. Violated tell dont ask

File: src/Api/Service/StatementSetter.php [hunk 3]
```diff
@@ -3,9 +3,9 @@
 namespace Wikibase\Api\Service;
 
 use InvalidArgumentException;
-use Mediawiki\Api\MediawikiApi;
-use Mediawiki\Api\SimpleRequest;
-use Wikibase\DataModel\Serializers\StatementSerializer;
```
Comment (JeroenDeDauw) [inline src/Api/Service/StatementSetter.php]: Bad @addshore, using package private classes!

**Comment by JeroenDeDauw** [comment]
+2 from my side

File: src/Api/Service/StatementSetter.php (hunk 3 above)
Comment (addshore) [inline src/Api/Service/StatementSetter.php]: Wahahahahaaaa, bad me indeed!

**Pull request merged**